src/modules/robot/Block.cpp

```diff
@@ -318,7 +320,32 @@ void Block::prepare(float acceleration_in_steps, float deceleration_in_steps)
     double deceleration_per_tick = deceleration_in_steps * fp_scale;
 
     for (uint8_t m = 0; m < n_actuators; m++) {
-        uint32_t steps = this->steps[m];
+        int32_t steps = this->steps[m];
+        double advance_velocity_offset = 0;
+        int8_t pressure_advance_direction = 1;
+
+        if(	(THEROBOT->actuators[m]->get_pressure_advance() != 0.0f) &&
+        	(this->primary_axis == true) &&	// Ignore extruder only moves
```

Comment:
also checking for pressure_advance == 0.0 is bogus due to IEEE floating point, it needs to be a range check.